frontend/test/components/bulkAssignProcedure.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { fireEvent, render, screen } from "@testing-library/vue"
+import { MockInstance } from "vitest"
+import BulkAssignProcedure from "@/components/BulkAssignProcedure.vue"
+import { PublicationState } from "@/domain/publicationStatus"
+import DocumentUnitService from "@/services/documentUnitService"
+import { ServiceResponse } from "@/services/httpClient"
+
+const AssignProcedureStub = {
+  template: `<span @click="$emit('assign-procedure', {label: 'Vorgangsname'})">AssignProcedureMock</span>`,
+}
+
+const addToastMock = vi.fn()
+vi.mock("primevue/usetoast", () => ({
+  useToast: () => ({ add: addToastMock }),
+}))
+
+describe("BulkAssignProcedure", () => {
+  let bulkAssignServiceSpy: MockInstance<
+    (
+      procedureLabel: string,
+      documentationUnitIds: string[],
+    ) => Promise<ServiceResponse<unknown>>
+  >
+  beforeEach(() => {
+    bulkAssignServiceSpy = vi.spyOn(DocumentUnitService, "bulkAssignProcedure")
+  })
+
+  it("should emit an error when no docUnits are selected", async () => {
+    const { emitted } = render(BulkAssignProcedure, {
+      props: { documentationUnits: undefined },
+      global: { stubs: { AssignProcedure: AssignProcedureStub } },
+    })
+
+    await fireEvent.click(screen.getByText("AssignProcedureMock"))
+
+    expect(emitted().updateSelectionErrors[0]).toEqual([
+      "Wählen Sie mindestens eine Dokumentationseinheit aus",
+      [],
+    ])
+    expect(bulkAssignServiceSpy).not.toHaveBeenCalled()
+  })
+
+  it("should emit an error when an empty list of docUnits is selected", async () => {
+    const { emitted } = render(BulkAssignProcedure, {
+      props: { documentationUnits: [] },
+      global: { stubs: { AssignProcedure: AssignProcedureStub } },
+    })
+
+    await fireEvent.click(screen.getByText("AssignProcedureMock"))
+
+    expect(emitted().updateSelectionErrors[0]).toEqual([
+      "Wählen Sie mindestens eine Dokumentationseinheit aus",
+      [],
+    ])
+    expect(bulkAssignServiceSpy).not.toHaveBeenCalled()
+  })
```

Comment:
Gibt es auch die Möglichkeit, dass es mehrere Errors gleichzeitig gibt? Wie geht die UI damit um?

Reply:
Also momentan gibt es eine feste Reihenfolge, in der die errors aufkommen können
1) Kein Vorgang ausgewählt 2) Keine Dok-Einheit ausgewählt 3) Eine unbestätigte Fremdanlage ausgewählt ...Request... 4) Request fehlgeschlagen

Die einzigen errors, die zusammen auftreten könnten wären 1) + 2) oder 1) + 3), so ist es auch im Design. Aber dann müsste man den error für die Vorgang selection im parent halten. Finde ich bisschen übertrieben dafür, dass das vermutlich nie auftritt. 🤔  Oder siehst du das anders?

Reply:
nein, macht total sinn, wir lassen es so :) war nur laut gedacht